### ISSUE TITLE
The anchors don't work

### ISSUE THREAD

**Issue by barbara79** [doxygen/doxygen#11135]
**Describe the bug**
 In my README file I have this
 `- [Introduction](#introduction)`
 
 and then
 ```
<a id="introduction"></a>
## Introduction
```
the behavior in the generated file via Doxygen is
```
<p><a class="anchor" id="introduction"></a> </p>
<h1><a class="anchor" id="autotoc_md2"></a>
Introduction</h1>
```
until 1 month ago it was working perfectly.
As you can imagine now it doesn't anchor properly so the list of indexes is not working.

**Screenshots**

<img width="332" alt="Screenshot 2024-09-12 at 17 51 37" src="https://github.com/user-attachments/assets/681799b5-5b65-44e6-92ff-bcd93d809e86">

**Expected behavior**
The anchors should work as expected with the names given by me
**Version**
1.12.0

**Stack trace**

**Additional context**
Just please let me know if you need something else like the Doxyfile

**Comment by albert-github** [comment]
It would indeed help when having a small complete example with the used settings, so
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

Furthermore you wrote:
> until 1 month ago it was working perfectly.

what change did you make, update of doxygen ? In that case from which version.


As a side note, the
```
<a id="introduction"></a>
```
is not necessary as there exists the setting: `MARKDOWN_ID_STYLE` that can generate markdown like anchors when the setting is `GITHUB` (the default is `Doxygen`). There are other ways of specifying anchors as well.

**Comment by barbara79** [comment]
Hello,
I prepared an example, I attached the .HTML file, but probably you don't need it, it's just share with you what I see after the generation.

[doxygen_prj.zip](https://github.com/user-attachments/files/16991110/doxygen_prj.zip)

**Comment by barbara79** [comment]
Just saw the MARKDOWN_ID_STYLE configuration ;) 
I removed the tag <a> but it doesn't work as well

**Comment by albert-github** [comment]
I think setting `MARKDOWN_ID_STYLE  = GITHUB` should work, at least it does for me.
I see there is a difference between the 1.11.0 and 1.12.0 version around the the beginning of July 2024 that looks like to cause the problems, bisecting at the moment.

**Comment by barbara79** [comment]
thanks, now the links are working, but it still doesn't show the logo and the project title correctly formatted
I still find  this behaviour:
`Loading...
Searching...
No Matches`

**Comment by barbara79** [comment]
if I inspect the page I see that it's before the header class

**Comment by albert-github** [comment]
The problem with "the logo and the project title correctly formatted" I don't get and with the provided example it is not reproducible either as not all files mentioned there are provided, like:
```
PROJECT_LOGO           = ct-logo.png
PROJECT_ICON           = ct-logo.png
LAYOUT_FILE            = DoxygenLayout.xml
HTML_HEADER            = header.html
HTML_EXTRA_STYLESHEET  = doxygen_extra.css
```
also the provided `index.html` brings nothing for me as it is not doxygen output but just a redirector file.
Also I think the issue with the "the logo and the project title correctly formatted" is a separate issue and a new issue should be opened for it.

**Comment by albert-github** [comment]
The result of the bisecting of the original anchor problem gives:
300f73d9ecd6b010a160e5f8cef14ecbabab950b is the first bad commit
```
commit 300f73d9ecd6b010a160e5f8cef14ecbabab950b

Date:   Thu Jul 4 20:41:56 2024 +0200

    issue #10970 Improvement of end of paragraph detection

 src/commentscan.l | 101 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-------------------------
 src/docnode.cpp   |  15 +++++++++++++++
 2 files changed, 91 insertions(+), 25 deletions(-)
```

**Comment by barbara79** [comment]
> The problem with "the logo and the project title correctly formatted" I don't get and with the provided example it is not reproducible either as not all files mentioned there are provided, like:
> 
> ```
> PROJECT_LOGO           = ct-logo.png
> PROJECT_ICON           = ct-logo.png
> LAYOUT_FILE            = DoxygenLayout.xml
> HTML_HEADER            = header.html
> HTML_EXTRA_STYLESHEET  = doxygen_extra.css
> ```
> 
> also the provided `index.html` brings nothing for me as it is not doxygen output but just a redirector file. Also I think the issue with the "the logo and the project title correctly formatted" is a separate issue and a new issue should be opened for it.

Although I believe that both problems are connected, I'm gonna open another issue.
I think it's about the new changes because as said before it was working perfectly.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #11140

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

(Still waiting for the new issue regarding "the logo and the project title correctly formatted")

**Comment by barbara79** [comment]
@albert-github I won't open it since like this it looks like ok.
Please close the issue once it will release a new version.
Thanks a lot for everything

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
